pkg/reporter/csv.go

```diff
@@ -0,0 +1,92 @@
+package reporter
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+
+	"github.com/safedep/vet/pkg/analyzer"
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/policy"
+	"github.com/safedep/vet/pkg/readers"
+)
+
+type csvReporter struct {
+	csvRecords []CsvRecord
+}
+
+type CsvRecord struct {
+	packageName string
+	updateTo    string
+}
+
+func NewCsvReporter() (Reporter, error) {
+	return &csvReporter{}, nil
+}
+
+func (r *csvReporter) Name() string {
+	return "Csv Report Generator"
+}
+
+func (r *csvReporter) AddManifest(manifest *models.PackageManifest) {
+	r.csvRecords = make([]CsvRecord,0)
+	readers.NewManifestModelReader(manifest).EnumPackages(func(pkg *models.Package) error {
```

Comment:
What are we trying to achieve here? Apologies for not writing the requirement clearly in the Github issue.

I am not sure if we should add every package into the CSV output because it will have too much data (aka. noise). How about we only add packages that match a filter into CSV output? For example:

```
vet scan -D /path --filter 'vulns.critical.exists(p, true)' --report-csv=/tmp/critical-vuln-packages.csv
```

See example here
https://github.com/safedep/vet/blob/main/pkg/reporter/markdown.go#L77

Thoughts?

Reply:
@abhisek Will change it to Path. 
So, it should be similar to the markdown report built on top of the summary reporter. Please correct me if I am wrong. 
What all fields should be present in CSV file headers?

Reply:
You can consider building the CSV reporter on top of summary reporter. I haven't considered the possibility but seems like it is an option because summary reporter already handles recording of events raised by Analyzers (Filter). 

I would consider the following fields in CSV to start with:

```
ecosystem, manifest path, packageName, packageVersion, filter violation reason (available from the event)
```

@c0d3G33k Any suggestion on what all fields will be useful in CSV report?

Reply:
@abhisek Got it. 
I am assuming we generate the CSV records only for violations as only they will have violation reason listed. 
Is my assumption correct ? 
https://github.com/safedep/vet/blob/main/pkg/reporter/markdown.go#L134

Reply:
@abhisek Pushed the changes. Please review